samples/05.custom-components/k.per-message-avatar/README.md

```diff
@@ -0,0 +1,375 @@
+-  Run `npx serve`
+-  Browse to [http://localhost:5000/](http://localhost:5000/)
+
+# Things to try out
+
+-  Watch the messages appear from Mock Bot.
+
+# Code
+
+> Jump to [completed code](#completed-code) to see the end-result `index.html`.
+
+## Overview
+
+> This sample is based on the [01.getting-started/e.host-with-react](https://github.com/microsoft/BotFramework-Webchat/tree/master/samples/01.getting-started/e.host-with-react).
+
+This sample is separated into 2 phases:
+
+-  Adding online status
+-  Rendering a custom avatar
+
+### Creating an online status component
+
+First, create a React component, `<AvatarWithOnlineStatus>` that will decorate its children with an online status icon.
+
+The component supports two online status: `"online"` and `"busy"`.
+
+```diff
+  const res = await fetch('https://webchat-mockbot.azurewebsites.net/directline/token', { method: 'POST' });
+  const { token } = await res.json();
+  const { ReactWebChat } = window.WebChat;
+
++ const AvatarWithOnlineStatus = ({ children, onlineStatus }) => {
++   return (
++     <div>
++       {children}
++       {onlineStatus && (
++         <div
++           className={`app__avatarWithOnlineStatus__status app__avatarWithOnlineStatus__status--${onlineStatus}`}
++         />
++       )}
++     </div>
++   );
++ };
+
+  window.ReactDOM.render(
+    <ReactWebChat directLine={window.WebChat.createDirectLine({ token })} />,
+    document.getElementById('webchat')
+  );
+```
+
+Then, add the stylesheet for the online status icon:
+
+```css
+.app__avatarWithOnlineStatus {
+   display: flex;
+   position: relative;
+}
+
+.app__avatarWithOnlineStatus .app__avatarWithOnlineStatus__status {
+   background-color: White;
+   border-radius: 50%;
+   border: solid 2px White;
+   bottom: -2px;
+   height: 10px;
+   position: absolute;
+   right: -2px;
+   transition: background-color 200ms;
+   width: 10px;
+}
+
+.app__avatarWithOnlineStatus .app__avatarWithOnlineStatus__status.app__avatarWithOnlineStatus__status--online {
+   background-color: #090;
+}
+
+.app__avatarWithOnlineStatus .app__avatarWithOnlineStatus__status.app__avatarWithOnlineStatus__status--busy {
+   background-color: Red;
+}
+```
+
+### Create an avatar middleware
+
+Then, creates an avatar middleware that will render `<AvatarWithOnlineStatus>` on top of the original avatar.
+
+`avatarMiddleware` should only return a function or `false`. If the avatar is hidden, it should return `false`. Otherwise, it should return a function, which when called, will render the avatar.
+
+In the following sample, `renderAvatar` is the original avatar of type `false | () => React.Element`. If the avatar should not be showed, `renderAvatar` will be `false` and the middleware return it. Otherwise, `renderAvatar` will be `() => React.Element`. The middleware will return `<AvatarWithOnlineStatus>` with the result of calling `renderAvatar()`.
```

Comment:
```suggestion
In the following sample, `renderAvatar` is the original avatar of type `false | () => React.Element`. If the avatar should not be shown, `renderAvatar` will be `false` and the middleware return it. Otherwise, `renderAvatar` will be: `() => React.Element`. The middleware will return `<AvatarWithOnlineStatus>` with the result of calling `renderAvatar()`.
```